src/network/wire/lib.zig

```diff
@@ -191,12 +189,36 @@ test "ok_send_mempool_message" {
 
     switch (received_message) {
         .Mempool => {},
-        .Verack => unreachable,
-        .Version => unreachable,
-        .Getaddr => unreachable,
+        else => unreachable,
     }
 }
 
+test "ok_send_ping_message" {
+    const Config = @import("../../config/config.zig").Config;
+    const ArrayList = std.ArrayList;
+    const test_allocator = std.testing.allocator;
+    const PingMessage = protocol.messages.PingMessage;
+
+    var list: std.ArrayListAligned(u8, null) = ArrayList(u8).init(test_allocator);
+    defer list.deinit();
+
+    const message = PingMessage.new(21000000);
+
+    const writer = list.writer();
+    try sendMessage(test_allocator, writer, Config.PROTOCOL_VERSION, Config.BitcoinNetworkId.MAINNET, message);
+    var fbs: std.io.FixedBufferStream([]u8) = std.io.fixedBufferStream(list.items);
+    const reader = fbs.reader();
+
+    const received_message = try receiveMessage(test_allocator, reader);
+    defer received_message.deinit(test_allocator);
+
+    switch (received_message) {
+        .Ping => {},
```

Comment:
.Ping => |ping_message| {try std.testing.expectEqual(message.nonce, ping_message.nonce}),

and you can remove your last line